### PR TITLE
Kinetic Crusher

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Implements the Kinetic Crusher, from TG: https://github.com/tgstation/tgstation/pull/19891

The Kinetic Crusher is a two-handed ***required*** melee weapon. It has a base damage of 20, making it a bit more effective at dealing with hostile mining wildlife than a pickaxe.

In addition to being a decent melee weapon, it has a few special abilities: When in a lower pressure environment, it can fire a limited range projectile that will mine out mining tiles. This projectile does no damage, but has a special property; when it hits a **large** mob (all mining wildlife counts as large), it will mark that mob, making the next melee strike against that mob by the kinetic smasher deal 50 bonus damage (total of 70). If you manage to strike a marked mob from behind, then the smasher will deal 80 bonus damage (total of 100). This marking effect cannot be applied to human sized or smaller mobs.

You can pick up the smasher with a voucher or purchase one, outright, for 750 mining points.

related:
- Fixes up the quick_equip proc to be more OOP
- Mobs can have different mob sizes now, instead of being governed purely by the mob var `small`
- check_target_facings no longer uses magic numbers and is far more accurate at checking when one atom is truly behind another

Fixes
- Fixes the Vox simple animal existing

:cl: Fox McCloud
add: Adds in the Kinetic Smasher as a mining purchase
/:cl:

### CODE REVIEW

File: code/modules/mob/mob.dm [hunk 1]
```diff
@@ -866,7 +866,10 @@ var/list/slot_equipment_priority = list( \
 /mob/MouseDrop(mob/M as mob)
 	..()
 	if(M != usr) return
-	if(M.small) return // Stops pAI drones and small mobs (borers, parrots, crabs) from stripping people. --DZD
+	if(isliving(M)) // Ewww
```
Comment (Fox-McCloud) [inline code/modules/mob/mob.dm]: This is gross, but unfortunately you can't do an override of this for living without causing horrific runtimes and breaking `MouseDrop` everywhere.

**Comment by KasparoVy** [comment]
God bless man this is such a dope piece of equipment. Can't wait to put it to work.

**Comment by Necaladun** [comment]
Fucking cool. Are their other large mobs? Queens, hulks, space pods, etc?

**Comment by KingPhilipIII** [comment]
I think for tg large mobs originally meant megafauna like Bubblegum or ash drakes. I haven't been playing tg for very long so I'm probably wrong.

**Comment by Fox-McCloud** [comment]
@KingPhilipIII 

They made all asteroid mobs "large".

@Necaladun 

The empresses are large; not many mobs are "large" though.

**Comment by Faryaa** [comment]
Looks awesome! Could you port explorer suit from /tg/ as well? Like alternative spacesuit, you can buy for points. Staticaly it would be identical to standart hardsuit, but fancier.

**Pull request merged**